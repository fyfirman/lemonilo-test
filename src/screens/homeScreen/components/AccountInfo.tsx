import Avatar from '@components/Avatar';
import React from 'react';
import styled from 'styled-components/native';
import AccountAction from './AccountAction';

const Container = styled.View`
  background-color: ${({ theme }) => theme.palette.background.grey};
  width: 100%;
  padding: ${({ theme }) => theme.spacing(1)}px ${({ theme }) => theme.spacing(2)}px;
`;

const InfoContainer = styled.View`
  flex-direction: row;
  align-items: center;
  padding-top: ${({ theme }) => theme.spacing(1)}px;
`;

const NameText = styled.Text`
  font-size: 16px;
  font-weight: bold;
  color: ${({ theme }) => theme.palette.text.primary};
  margin-left: ${({ theme }) => theme.spacing(1)}px;
`;

const AccountInfo = () => {
  return (
    <Container>
      <InfoContainer>
        <Avatar source={require('@assets/images/avataaars.png')} />
        <NameText>Firmansyah Yanuar</NameText>
      </InfoContainer>
      <InfoContainer>
        <AccountAction
          label="My Points"
          onPress={() => console.log('Account actions')}
          style={{ marginRight: 8 }}
        />
        <AccountAction label="Payment" onPress={() => console.log('Account actions')} />
      </InfoContainer>
    </Container>
  );
};

export default AccountInfo;
