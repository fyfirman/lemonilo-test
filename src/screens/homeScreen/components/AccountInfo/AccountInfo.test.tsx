import React from 'react';
import { renderWithTheme } from '../../../../__tests__/helpers';
import AccountInfo from './AccountInfo';
import 'jest-styled-components';

test('should match with snapshot', () => {
  const tree = renderWithTheme(<AccountInfo />);

  expect(tree).toMatchSnapshot();
});
