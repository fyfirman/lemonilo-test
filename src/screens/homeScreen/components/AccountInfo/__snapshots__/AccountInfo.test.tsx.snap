// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F5F9F8",
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingTop": 8,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/avataaars.png",
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "#19A0E0",
            "borderRadius": 100,
            "height": 40,
            "width": 40,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "rgba(0,0,0,0.8)",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginLeft": 8,
          },
        ]
      }
    >
      Firmansyah Yanuar
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingTop": 8,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "paddingBottom": 4,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 4,
          },
          Object {
            "marginRight": 8,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(0,0,0,0.8)",
              "fontSize": 11,
              "fontWeight": "bold",
            },
          ]
        }
      >
        My Points
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "paddingBottom": 4,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 4,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(0,0,0,0.8)",
              "fontSize": 11,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Payment
      </Text>
    </View>
  </View>
</View>
`;
